src/main/java/com/google/api/generator/gapic/composer/ServiceSettingsClassComposer.java

```diff
@@ -119,11 +120,21 @@ public GapicClass generate(Service service, Map<String, Message> ignore) {
     return GapicClass.create(kind, classDef);
   }
 
-  private static List<CommentStatement> createClassHeaderComments(Service service) {
+  private static List<CommentStatement> createClassHeaderComments(
+      Service service, TypeNode classType) {
+    // Pick the first pure unary rpc method, if no such method exist, then pick the first in the
+    // list.
     Optional<Method> methodOpt =
-        service.methods().isEmpty() ? Optional.empty() : Optional.of(service.methods().get(0));
+        service.methods().isEmpty()
+            ? Optional.empty()
+            : Optional.of(
+                service.methods().stream()
+                    .reduce(
```

Comment:
Since the intent of this is to get some element that matches the criteria (and not a combination, which reduction would be better suited for), consider getting the first element out of  `filter` instead.

Also, what is the expected behavior if there are no methods that match this criteria?

Reply:
It is common to use `reduce` to get the first matches the criteria element. Here is what I got: https://www.geeksforgeeks.org/find-the-first-element-of-a-stream-in-java/

If no matched element exist, we use the first method in the methods list. See videointelligence, it use the first method where it is not the pure unary method.
https://github.com/googleapis/java-video-intelligence/blob/master/google-cloud-video-intelligence/src/main/java/com/google/cloud/videointelligence/v1/VideoIntelligenceServiceSettings.java#L55-L63

Reply:
Would prefer `findFirst` over `reduce` because it's more semantically consistent.

Reply:
sg